network.go

```diff
@@ -52,6 +60,27 @@ func (network *network) SetNetwork(net map[string]AccountID) error {
 func (network *network) getNodeAccountIDsForExecute() []AccountID {
 	sort.Sort(nodes{nodes: network.nodes})
 
+	if network.maxNodeAttempts > 0 {
+		for i := 0; i < len(network.nodes); i++ {
+			var nod *node
+			if network.nodes[i] != nil {
+				nod = network.nodes[i]
+			} else {
+				continue
+			}
+			if nod.attempts >= int64(network.maxNodeAttempts) {
+				err := nod.close()
+				if err != nil {
+					panic(err)
+				}
+				network.nodes = append(network.nodes[:i], network.nodes[i+1:]...)
```

Comment:
I just realized that this is how you're supposed to delete elements from a list in Go :vomiting_face: